### PR TITLE
Added oracle entry contract to SDKv2

### PR DESCRIPTION
**Pull request by consvic**
## Changes
- Added oracle entry contract to sdk v2
- Created `getOraclePrice` function to fetch prices from oracle contract

### CODE REVIEW

File: src/sdkV2.ts [hunk 1]
```diff
@@ -53,36 +49,53 @@ export default class CegaEvmSDKV2 {
     return chainConfig;
   }
 
-  loadCegaEntry(): ethers.Contract {
+  loadAddressManager(): ethers.Contract {
     return new ethers.Contract(
-      this._cegaEntryAddress,
+      this._addressManagerAddress,
```
Comment (vsai) [inline src/sdkV2.ts:54]: to `loadAddressManager` you need to use that ABI

File: src/sdkV2.ts [hunk 2]
```diff
@@ -53,36 +49,53 @@ export default class CegaEvmSDKV2 {
...
+    return new ethers.Contract(
+      oracleEntryAddress,
+      IDCSEntryAbi.abi,
+      this._signer || this._provider,
+    );
+  }
+
   async dcsGetProduct(productId: ethers.BigNumberish) {
-    const cegaEntry = this.loadCegaEntry();
+    const cegaEntry = await await this.loadCegaEntry();
```
Comment (vsai) [inline src/sdkV2.ts:90]: shouldn't need 2 awaits

File: src/sdkV2.ts [hunk 3]
```diff
@@ -111,8 +124,10 @@ export default class CegaEvmSDKV2 {
     asset: EvmAddress,
     overrides: TxOverrides = {},
   ): Promise<ethers.providers.TransactionResponse> {
+    const addressManager = this.loadAddressManager();
+    const cegaEntryAddress = await addressManager.getCegaEntry();
```
Comment (vsai) [inline src/sdkV2.ts:128]: nit:
```
const cegaEntry = await this.loadCegaEntry()
```
then use `cegaEntry.address`

Allows us to reduce access to the address manager when not needed

File: src/sdkV2.ts [hunk 4]
```diff
@@ -123,8 +138,13 @@ export default class CegaEvmSDKV2 {
     asset: EvmAddress,
     overrides: TxOverrides = {},
   ): Promise<ethers.providers.TransactionResponse> {
+    const addressManager = this.loadAddressManager();
+    const chainConfig = await this.getChainConfig();
+    const cegaWrappingProxyAddress = await addressManager.getAssetWrappingProxy(
```
Comment (vsai) [inline src/sdkV2.ts:143]: similar here:
```
const cegaWrappingProxy = await this.loadCegaWrappingProxy()
```
then `cegaWrappingProxy.address`

File: src/testScripts/sdkV2-scripts.ts [hunk 5]
```diff
@@ -17,16 +17,14 @@ const ADMIN_ACCOUNTS = {
 const CONFIGS = {
   ethereum: {
     RPC_URL: process.env.ETH_RPC_URL,
-    cegaEntryAddress: '0x6891F6594Ec545077561Cc4Fa79a861db2cfBA9D' as types.EvmAddress,
-    cegaWrappingProxyAddress: '0xd940d163b1C29F8a8bFc7dd6300a94fF47AaDECB' as types.EvmAddress,
+    addressManager: '' as types.EvmAddress,
```
Comment (vsai) [inline src/testScripts/sdkV2-scripts.ts:20]: use the address manager addresses here for ETH & ARB
https://www.notion.so/cega/QA-DCS-Deployments-a1d8fd0a7a304892a0b25875a9a79784

File: src/sdkV2.ts (hunk 2 above)
Comment (consvic) [reply src/sdkV2.ts]: whoopsie, updated

File: src/sdkV2.ts (hunk 3 above)
Comment (consvic) [reply src/sdkV2.ts]: ahhh nicee yeh that's better

File: src/testScripts/sdkV2-scripts.ts (hunk 5 above)
Comment (consvic) [reply src/testScripts/sdkV2-scripts.ts]: added addresses from notion

File: src/sdkV2.ts (hunk 1 above)
Comment (consvic) [reply src/sdkV2.ts:56]: updated ABI :)

**Comment by consvic** [comment]
@vsai addressed changes from PR 😄

**Review by vsai — APPROVED** [review]

**Pull request merged**